### PR TITLE
update queue to show queue in the modal and added custom url download

### PR DESCRIPTION
**Pull request by arslan2012**

### CODE REVIEW

File: service/model_manager/model_installer.py [hunk 1]
```diff
@@ -56,32 +56,29 @@ def parse_wget_output(output):
...
-        download_url_with_agent(url, save_path)
-
-        # Indicate that the task is done
-        download_queue.task_done()
-
-    # If all tasks are done, reset the progress and exit
-    with download_progress:
-        download_progress.value['progress'] = ""
-        download_progress.notify_all()
+    while True:
```
Comment (Weixuanf) [inline service/model_manager/model_installer.py:67]: i wonder this compared to `while not download_tasks.empty()`

File: service/model_manager/model_installer.py [hunk 2]
```diff
@@ -6,9 +6,9 @@
 from aiohttp import web
 import folder_paths
 import urllib.request
+import threading
 import server
-import server
-from .missing_models import find_missing_models
```
Comment (Weixuanf) [inline service/model_manager/model_installer.py:11]: 这里面有我missing node需要的code和endpoint，find_missing_models，

File: ui/src/model-manager/install-models/InatallModelsModal.tsx [hunk 3]
```diff
@@ -92,139 +101,175 @@ export default function InatallModelsModal({
...
+              })}
+            </HStack>
+            <Stack spacing={5} pos="absolute" bottom="0" left="0" width="50%" zIndex={80} backgroundColor="white" paddingX={5}>
+              {queue.map(({ save_path, progress }) => (
+                <HStack>
+                  <Text fontSize={16} width="40%">{save_path.replace(/^.*[\\/]/, '')}</Text>
+                  <Progress isIndeterminate={!progress} hasStripe width="60%" value={progress} />
+                </HStack>
+              ))}
+            </Stack>
```
Comment (Weixuanf) [inline ui/src/model-manager/install-models/InatallModelsModal.tsx:267]: 这个最好还是写在单独的InstallProgress里，split component to smaller chunks is a better for readability and reusing

**Comment by Weixuanf** [comment]
Screenshot 2024-01-18 at 3 11 30 PM
current bundle

File: service/model_manager/model_installer.py (hunk 1 above)
Comment (arslan2012) [reply service/model_manager/model_installer.py:67]: `Queue` is FIFO, so it's impossible to show the queue to the user, so I had to switch to `List`, which doesn't have `empty()` method

**Comment by Weixuanf** [comment]
List.size == 0 is the same? Not sure about the syntax but just an example

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by Weixuanf** [comment]
Not wrong! Just curious! We can keep this is as it is! Thanks for the good work

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Pull request merged**